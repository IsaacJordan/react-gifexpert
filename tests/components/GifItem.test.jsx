import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe('Probando el componente <GifItem />', () => { 

    const title = 'Saitama';
    const url = 'https://one-punch.com/'

    test('evaluacion del snapshot', () => {
        
        const {container} = render(<GifItem title={title} url={url} />);
        expect ( container ).toMatchSnapshot();

    });

    test('debe de mostrar la imagen con el url indicado', () => {

        render( <GifItem title={title} url={url} />);
        //screen.debug();
        //expect( screen.getByRole('img').src).toBe( url );
        const {src, alt} = screen.getByRole('img');
        expect( src ).toBe( url );
        expect( alt ).toBe( alt );

    });

    test('debe de mostrar el titulo en el componente', () => {

        render( <GifItem title={ title } url={ url }/>)
        expect( screen.getByText( title )).toBeTruthy();

    });


});