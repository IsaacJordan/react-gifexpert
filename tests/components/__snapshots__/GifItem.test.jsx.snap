// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando el componente <GifItem /> evaluacion del snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="https://one-punch.com/"
    />
    <p>
      Saitama
       
    </p>
  </div>
</div>
`;
